packages/server-wallet/src/wallet/wallet.ts

```diff
@@ -33,32 +38,73 @@ export class Wallet extends EventEmitter<ObjectiveProposed> {
    */
   public static async create(
     engine: Engine,
+    chainService: ChainServiceInterface,
     messageServiceFactory: MessageServiceFactory,
     retryOptions: Partial<RetryOptions> = DEFAULTS
   ): Promise<Wallet> {
-    return new Wallet(messageServiceFactory, engine, {...DEFAULTS, ...retryOptions});
+    await chainService.checkChainId(engine.engineConfig.networkConfiguration.chainNetworkID);
+    const wallet = new Wallet(messageServiceFactory, engine, chainService, {
+      ...DEFAULTS,
+      ...retryOptions,
+    });
+    await wallet.registerExistingChannelsWithChainService();
+    return wallet;
   }
 
   private _messageService: MessageServiceInterface;
+  private _destroyed = false;
 
   private constructor(
     messageServiceFactory: MessageServiceFactory,
     private _engine: Engine,
+    private _chainService: ChainServiceInterface,
     private _retryOptions: RetryOptions
   ) {
     super();
+
     const handler: MessageHandler = async message => {
-      const {outbox, newObjectives} = await this._engine.pushMessage(message.data);
+      const {outbox, newObjectives, channelResults} = await this._engine.pushMessage(message.data);
       // Receiving messages from other participants may have resulted in new proposed objectives
       for (const o of newObjectives) {
         this.emit('ObjectiveProposed', o);
+
+        // If this is a new open channel objective that means there is a new channel to be monitored in the chain service
+        if (o.type === 'OpenChannel') {
+          const listener = createChainListener(
```

Comment:
Do we need to create a new instance of a chain listener per channel? Or can one instance of a chain listener get used for all channels?

Reply:
Good point! Andrew also flagged that in his [review](https://github.com/statechannels/statechannels/pull/3564#discussion_r647036909)

Reply:
For merging simplicity I've just made this change upstream in 19a3e2dbedcd9153b142f49d3f208ce009e6ceb1